principles.md

```diff
@@ -0,0 +1,31 @@
+# usethis design principles
+
+*This is an experiment in making key package design principles explicit, versus only implicit in the code. The goal is to make maintenance easier, when spread out over time and across people.*
+
+## Active project
+
+Many usethis functions act on the **active project**, the path to which is stored in the internal environment `proj`, specifically in `proj$cur`. We do this instead of constantly passing around a base path. It is implied that such functions create or modify files inside the active project. This is mostly true of `use_*()` functions, though there are exceptions. For example, `use_course()` makes no reference to the active project.
```

Comment:
We do this instead of constantly passing around a base path or relying on the working directory.